pkg/maps/ctmap/ctmap.go

```diff
@@ -45,48 +45,27 @@ var (
 		metrics.LabelDatapathFamily: "ipv4",
 	}
 
-	mapInfo = map[MapType]struct {
-		keySize    int
-		maxEntries int
-		parser     bpf.DumpParser
-		bpfDefine  string
-	}{
-		MapTypeIPv4Local: {
-			keySize:    int(unsafe.Sizeof(CtKey4{})),
-			maxEntries: MapNumEntriesLocal,
-			parser:     ct4DumpParser,
-			bpfDefine:  "CT_MAP4",
-		},
-		MapTypeIPv6Local: {
-			keySize:    int(unsafe.Sizeof(CtKey6{})),
-			maxEntries: MapNumEntriesLocal,
-			parser:     ct6DumpParser,
-			bpfDefine:  "CT_MAP6",
-		},
-		MapTypeIPv4Global: {
-			keySize:    int(unsafe.Sizeof(CtKey4{})),
-			maxEntries: MapNumEntriesGlobal,
-			parser:     ct4DumpParser,
-			bpfDefine:  "CT_MAP4",
-		},
-		MapTypeIPv6Global: {
-			keySize:    int(unsafe.Sizeof(CtKey6{})),
-			maxEntries: MapNumEntriesGlobal,
-			parser:     ct6DumpParser,
-			bpfDefine:  "CT_MAP6",
-		},
-	}
+	mapInfo = make(map[MapType]mapAttributes)
 )
 
 const (
 	// mapCount counts the maximum number of CT maps that one endpoint may
 	// access at once.
-	mapCount = 2
-
-	MapName6       = "cilium_ct6_"
-	MapName4       = "cilium_ct4_"
-	MapName6Global = MapName6 + "global"
-	MapName4Global = MapName4 + "global"
+	mapCount = 4
+
+	// Map names for TCP CT tables are retained from Cilium 1.0 naming
+	// scheme to minimize disruption of ongoing connections during upgrade.
+	MapNamePrefix     = "cilium_ct"
+	MapNameTCP6       = MapNamePrefix + "6_"
+	MapNameTCP4       = MapNamePrefix + "4_"
+	MapNameTCP6Global = MapNameTCP6 + "global"
+	MapNameTCP4Global = MapNameTCP4 + "global"
+
+	// Map names for "any" protocols indicate CT for non-TCP protocols.
+	MapNameAny6       = MapNamePrefix + "_any6_"
+	MapNameAny4       = MapNamePrefix + "_any4_"
+	MapNameAny6Global = MapNameAny6 + "global"
+	MapNameAny4Global = MapNameAny4 + "global"
 
 	MapNumEntriesLocal  = 64000
 	MapNumEntriesGlobal = 1000000
```

Comment:
I suggest we do a 600K TCP and 400K UDP split going to get started and start running that to see how it goes. We'll likely have to make it configurable in the end but that change is a bit more invasive.

Reply:
Seems reasonable. If we do this, we will force CT map recreation on upgrade so might as well name them better on the filesystem too. (The alternative would involve leaving the existing map's details the same and only use it for TCP connections, which should allow upgrade/downgrade to persist the connections without interruption).